src/core/cgroup.c

```diff
@@ -3034,6 +3058,11 @@ static int cg_bpf_mask_supported(CGroupMask *ret) {
         if (r > 0)
                 mask |= CGROUP_MASK_BPF_DEVICES;
 
+        /* BPF pinned prog */
+        r = cg_all_unified();
+        if (r > 0)
+                mask |= CGROUP_MASK_BPF_FOREIGN;
```

Comment:
@poettering 

The feature implies that BPF program is already loaded into kernel externally, hence BPF is available.
I'd check for is specific attach type but cgroup context where properties are stored is not available at the moment that code executes.
I can add probing for either egress or ingress hooks since among all cgroup hooks they were added first to the kernel.

Reply:
hmm, what about this simple test:

```
int bpf_foreign_supported(void) {
        r = cg_all_unified();
        if (r <= 0)
                return r;
        return path_is_mount_point("/sys/fs/bpf", NULL, 0);       
}
```

i.e. check if we are in unified mode *and* that bpffs is actually mounted. That should be enough for us to test, and is the right thing to do in containers and such where bpffs might be missing and where we hence should assume the bpf foreign stuff is not available.